sdk/core/core/inc/az_json_read.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_JSON_READ_H
+#define AZ_JSON_READ_H
+
+#include <az_const_str.h>
+#include <az_result.h>
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum {
+  AZ_JSON_NO_MORE_ITEMS = AZ_JSON_RESULT + 1,
+  AZ_JSON_ERROR_UNEXPECTED_END,
+  AZ_JSON_ERROR_UNEXPECTED_CHAR,
+  AZ_JSON_ERROR_INVALID_STATE,
+  AZ_JSON_ERROR_STACK_OVERFLOW,
+};
+
+typedef az_const_str az_json_string;
```

Comment:
I don't love typedefs like this, it makes you have to jump through more "go to definitions" to find the actual type, and it's not like you're ever going to be able to just swap it out.